langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/DefaultMcpClient.java

```diff
@@ -233,7 +235,10 @@ public String executeTool(ToolExecutionRequest executionRequest) {
         } catch (TimeoutException timeout) {
             transport.executeOperationWithoutResponse(new McpCancellationNotification(operationId, "Timeout"));
             return ToolExecutionHelper.extractResult(RESULT_TIMEOUT);
-        } catch (ExecutionException | InterruptedException e) {
+        } catch (ExecutionException e) {
+            throw new ToolExecutionException(e.getCause());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
This looks harmless, but I'm not sure if it is necessary, isn't a thread already interrupted when the `InterruptedException` is thrown?